api/features/feature_segments/limits.py

```diff
@@ -0,0 +1,52 @@
+from django.db.models import Q
+
+from environments.models import Environment
+from features.versioning.models import EnvironmentFeatureVersion
+
+SEGMENT_OVERRIDE_LIMIT_EXCEEDED_MESSAGE = (
+    "The environment has reached the maximum allowed segments overrides limit."
+)
+
+
+def exceeds_segment_override_limit(
+    environment: Environment,
+    segment_ids_to_create_overrides: list[int] = None,
+    segment_ids_to_delete_overrides: list[int] = None,
```

Comment:
I understand it's setting them equal to empty list below, but I still think the typing should match the default arguments.
```suggestion
    segment_ids_to_create_overrides: list[int] | None = None,
    segment_ids_to_delete_overrides: list[int] | None = None,
```

Reply:
Yep, that's a fair point, updated.